lib/RemoveInitiatorBehaviour.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var inherits = require('inherits');
+
+var CommandInterceptor = require('diagram-js/lib/command/CommandInterceptor').default;
+var is = require('bpmn-js/lib/util/ModelUtil').is;
+
+/**
+ * Remove 'camunda:initiator' property when a startEvent is moved to or created within a subProcess
+ */
+function RemoveInitiatorBehaviour(
+    modeling, injector
+) {
+
+  injector.invoke(CommandInterceptor, this);
+
+
+  this.postExecuted(['shape.create','shape.move'], function(context) {
+
+    var shape = context.shape;
+
+    if (is(shape, 'bpmn:StartEvent')) {
+
+      // if subProcess becomes the new parent
+      if (!is(context.oldParent, 'bpmn:SubProcess') && (is(context.newParent, 'bpmn:SubProcess'))) {
+
+        // remove initiator property
+        modeling.updateProperties(shape, { 'camunda:initiator': undefined });
+
+      }
+    }
```

Comment:
Let's be explicit here. The conditions for the hook are:
* The moved/created shape is a Start Event.
* The start event has `camunda:initiator` (even if it's an empty string).
* New parent is a SubProcess.

What we don't care about is the old parent. The diagram could have been broken but behavior still needs to kick in.
Also, in the current implementation we trigger an unnecessary update if the start event does not have `camunda:initiator`.

My suggestion: check all three conditions and use `modeling.updateProperties` only if all of them are met.

Reply:
So the test case which is missing is when you move a start event which has the property and is already a child of a subprocess.